dipy/reconst/rumba.py

```diff
@@ -0,0 +1,815 @@
+        fodf = np.maximum(f_zero, fodf)  # apply positivity constraint
+
+        # Update other variables
+        reblurred = np.matmul(kernel, fodf)
+        reblurred_s = data * reblurred / sigma2
+
+        # Iterate variance
+        sigma2_i = (1/(n_g * n_order)) * np.sum((data**2 + reblurred **
+                                                 2) / 2 - (sigma2 * reblurred_s) * ratio, axis=0)
+        sigma2_i = np.minimum((1/8)**2, np.maximum(sigma2_i, (1/80)**2))
+        sigma2 = sigma2_i * np.ones(data.shape)  # Expand into vector
+
+    fodf = fodf / (np.sum(fodf, axis=0) + EPS)  # normalize final result
+
+    fodf_wm = np.squeeze(fodf[:n_c-1])  # white matter components
+    f_csf = np.squeeze(fodf[n_c-1])  # CSF component
+
+    return fodf_wm, f_csf
+
+
+def mbessel_ratio(n, x):
+    '''
+    Fast computation of modified Bessel function ratio (first kind).
+
+    Computes:
+
+    $I_{n}(x) / I_{n-1}(x)$
+
+    using Perron's continued fraction equation where $I_n$ is the modified Bessel function of first
+    kind, order $n$ [1]_.
+
+    Parameters
+    ----------
+    n : int
+        Order of Bessel function in numerator (denominator is of order n-1). Must be a positive
+        int.
+    x : float or ndarray
+        value or array of values with which to compute ratio
+
+    Returns
+    -------
+    y : float or ndarray
+        Result of ratio computation.
+
+    References
+    ----------
+    .. [1] W. Gautschi and J. Slavik, “On the computation of modified Bessel function ratios,”
+           Math. Comp., vol. 32, no. 143, pp. 865–875, 1978, doi: 10.1090/S0025-5718-1978-0470267-9
+    '''
+
+    y = x / ((2*n + x) - (2*x * (n + 1/2) / (2*n + 1 + 2*x - (2*x * (n + 3/2) / (
+        2*n + 2 + 2*x - (2*x * (n + 5/2) / (2*n + 3 + 2*x)))))))
+
+    return y
+
+
+def generate_kernel(gtab, sphere, lambda1=1.7e-3, lambda2=0.2e-3, lambda_csf=3.0e-3):
+    '''
+    Generate deconvolution kernel
+
+    Compute kernel mapping volume fractions of white matter fiber populations (oriented along each
+    vertex of a sphere) and cerebrospinal fluid (CSF) to a diffusion weighted signal.
+
+    Parameters
+    ----------
+    gtab : GradientTable
+    sphere : Sphere
+        sphere with which to sample discrete fiber orientations in order to construct kernel
+    lambda1 : float, optional
+        white matter diffusivity parallel to fiber axis (first DTI eigenvalue). Default: 1.7e-3
+    lambda2 : float
+        white matter diffusivity perpendicular to fiber axis (second/third DTI eigenvalues are
+        assumed equal). Default: 0.2e-3
+    lambda_csf : float
+        Mean diffusivity for CSF. If `None`, then CSF kernel column set to all zeroes.
+        Default: 3.0e-3
+
+    Returns
+    -------
+    kernel : 2d ndarray (N, M)
+        computed kernel; can be multiplied with a vector consisting of volume fractions for each of
+        M-1 fiber populations and the fraction of CSF to produce a diffusion weighted signal.
```

Comment:
should we include a `predict.()` function that returns the modelled signal (similar to the DTI model)? @ejcanalesr would that be useful?